modules/core/entity/tests/src/Functional/FarmEntityBundleFieldTest.php

```diff
@@ -59,6 +59,51 @@ protected function setUp():void {
     $this->moduleInstaller = $this->container->get('module_installer');
   }
 
+  /**
+   * Test that bundle field maps are updated on install/uninstall.
+   */
+  public function testBundleFieldMapUpdates() {
+
+    // Get the entity field map.
+    $field_map = $this->entityFieldManager->getFieldMap();
+
+    // Confirm that the 'test_default_bundle_field' exists in the log field map.
+    $this->assertArrayHasKey('test_default_bundle_field', $field_map['log']);
+
+    // Confirm that the 'test_contrib_hook_bundle_field' does NOT exist (yet).
+    $this->assertArrayNotHasKey('test_contrib_hook_bundle_field', $field_map['log']);
+
+    // Install the farm_entity_contrib_test module.
+    $result = $this->moduleInstaller->install(['farm_entity_contrib_test']);
+    $this->assertTrue($result);
+
+    // Reload the entity field map. We need to get a new instance of the
+    // entity_field.manager service from the container without old state.
+    $this->container->set('entity_field.manager', NULL);
+    $this->entityFieldManager = $this->container->get('entity_field.manager');
+    $field_map = $this->entityFieldManager->getFieldMap();
+
+    // Confirm that the 'test_contrib_hook_bundle_field' exists in the log field
+    // map, and exists in the 'test' bundle, but not in 'test_override'.
+    $this->assertArrayHasKey('test_contrib_hook_bundle_field', $field_map['log']);
+    $this->assertContains('test', $field_map['log']['test_contrib_hook_bundle_field']['bundles']);
+    $this->assertNotContains('test_override', $field_map['log']['test_contrib_hook_bundle_field']['bundles']);
+
+    // Uninstall the farm_entity_contrib_test module.
+    $result = $this->moduleInstaller->uninstall(['farm_entity_contrib_test']);
+    $this->assertTrue($result);
+
+    // Reload the entity field map. We need to get a new instance of the
+    // entity_field.manager service from the container without old state.
+    $this->container->set('entity_field.manager', NULL);
+    $this->entityFieldManager = $this->container->get('entity_field.manager');
+    $field_map = $this->entityFieldManager->getFieldMap();
```

Comment:
Updated the patch for the core issue so that the ::rebuildBundleFieldMap() helper function resets the cache. In the tests we still need to reset the internal state of the `entity_field.manager` service though, otherwise we won't get new values. I think its safer to get a new instance from the container rather than call the `->setMap([])` helper method that seems like a weird thing to exist....

Reply:
LGTM thanks @paul121!